src/quacc/runners/ase.py

```diff
@@ -453,3 +459,188 @@ def _set_sella_kwargs(self, optimizer_kwargs: dict[str, Any]) -> None:
 
         if not self.atoms.pbc.any() and "internal" not in optimizer_kwargs:
             optimizer_kwargs["internal"] = True
+
+
+def run_neb(
+    images: list[Atoms],
+    relax_cell: bool = False,
+    fmax: float = 0.01,
+    max_steps: int | None = 1000,
+    optimizer: NEBOptimizer | Optimizer = NEBOptimizer,
+    optimizer_kwargs: dict[str, Any] | None = None,
+    run_kwargs: dict[str, Any] | None = None,
+    neb_kwargs: dict[str, Any] | None = None,
+    copy_files: SourceDirectory | dict[SourceDirectory, Filenames] | None = None,
+) -> Optimizer:
+    """
+    Run NEB optimization.
+
+    Parameters
+    ----------
+    images
+        List of images representing the initial path.
+    relax_cell
+        Whether to relax the unit cell shape and volume.
+    fmax
+        Tolerance for the force convergence (in eV/A).
+    max_steps
+        Maximum number of steps to take.
+    optimizer
+        Optimizer class to use. All Optimizers except BFGSLineSearch
+    optimizer_kwargs
+        Dictionary of kwargs for the optimizer.
+    run_kwargs
+        Dictionary of kwargs for the run() method of the optimizer.
+    neb_kwargs
+        Dictionary of kwargs for the NEB.
+    copy_files
+        Files to copy before running the calculation.
+
+    Returns
+    -------
+    Optimizer
+        The ASE Optimizer object.
+    """
+    if optimizer.__name__ == "BFGSLineSearch":
+        raise ValueError("BFGSLineSearch is not allowed as optimizer with NEB.")
+
+    # Copy atoms so we don't modify it in-place
+    images = copy_atoms(images)
+    settings = get_settings()
+
+    neb = NEB(images, **neb_kwargs)
+
+    dir_lists = []
+    # Perform staging operations
+    # this calc_setup function is not suited for multiple Atoms objects
+    for image in images:
+        tmpdir_i, job_results_dir_i = calc_setup(image, copy_files=copy_files)
+        dir_lists.append([tmpdir_i, job_results_dir_i])
+
+    # Set defaults
+    optimizer_kwargs = recursive_dict_merge(
+        {
+            "logfile": "-" if settings.DEBUG else dir_lists[0][0] / "opt.log",
+            "restart": dir_lists[0][0] / "opt.json",
+        },
+        optimizer_kwargs,
+    )
+    run_kwargs = run_kwargs or {}
+    traj_filename = "opt.traj"
+    # Check if trajectory kwarg is specified
+    if "trajectory" in optimizer_kwargs:
+        msg = "Quacc does not support setting the `trajectory` kwarg."
+        raise ValueError(msg)
+
+    # Define the Trajectory object
+    traj_file = dir_lists[0][0] / traj_filename
+    traj = Trajectory(traj_file, "w", atoms=neb)
+
+    # Set volume relaxation constraints, if relevant
+    for i in range(len(images)):
+        if relax_cell and images[i].pbc.any():
+            images[i] = FrechetCellFilter(images[i])
+
+    # Run optimization
+    dyn = optimizer(neb, **optimizer_kwargs)
+    dyn.attach(traj.write)
+    dyn.run(fmax, max_steps)
+    traj.close()
+
+    traj.filename = traj_file
+    dyn.trajectory = traj
```

Comment:
I moved it to after the clean up. Thanks for clarifying this part. This was before I realized that the tmp files are renamed to quacc-* files instead of getting deleted.